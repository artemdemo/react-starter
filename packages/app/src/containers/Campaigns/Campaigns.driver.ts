import { RenderResults } from '../../__tests__/testRender';
import { testIdExistsAsync } from '../../__tests__/utils';
import { TCampaign } from './Campaigns';

type CampaignOptions = {
  name?: string;
  picture?: string;
  description?: string;
};

export const createCampaignMock = (
  options: CampaignOptions = {},
): TCampaign => {
  const {
    name = 'test campaign',
    picture = 'http://via.placeholder.com/150x150',
    description = 'Est amet voluptate',
  } = options;
  return {
    name,
    picture,
    description,
  };
};

export class CampaignsDriver {
  constructor(public readonly component: RenderResults) {}

  async campaignsListExist(): Promise<boolean> {
    return testIdExistsAsync(this.component, 'campaigns-list');
  }

  async getCampaignsAmount(): Promise<number> {
    const { findAllByTestId } = this.component;
    const campaignsEl = await findAllByTestId('single-campaign');
    return campaignsEl.length;
  }
}
