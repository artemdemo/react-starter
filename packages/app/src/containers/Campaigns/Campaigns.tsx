import React from 'react';

export type TCampaign = {
  name: string;
  picture: string;
  description: string;
};

type TProps = {
  items?: TCampaign[];
};

export const Campaigns: React.FC<TProps> = (props) => {
  const items = props.items || [];
  return (
    <div className="row" data-testid="campaigns-list">
      {items.map((item) => (
        <div
          className="col-6 col-md-4"
          key={`campaigns-item ${item.name}`}
          data-testid="single-campaign"
        >
          <h3>{item.name}</h3>
          <img src={item.picture} alt={item.name} />
          <p>{item.description}</p>
        </div>
      ))}
    </div>
  );
};
