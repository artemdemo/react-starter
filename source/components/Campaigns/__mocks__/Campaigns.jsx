import React from 'react';

const Campaigns = props => (
    <div data-mock='Campaigns'>
        {JSON.stringify(props, null, 2)}
    </div>
);

export default Campaigns;
