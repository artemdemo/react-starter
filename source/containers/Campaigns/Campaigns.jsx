import React from 'react';
import PropTypes from 'prop-types';

const Campaigns = (props) => {
    return (
        <div className='row'>
            {props.items.map(item => (
                <div
                    className='col-6 col-md-4'
                    key={`campaigns-item ${item.name}`}
                >
                    <h3>{item.name}</h3>
                    <img
                        src={item.picture}
                        alt={item.name}
                    />
                    <p>
                        {item.description}
                    </p>
                </div>
            ))}
        </div>
    );
};

Campaigns.propTypes = {
    items: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        picture: PropTypes.string,
        description: PropTypes.string,
    })),
};

Campaigns.defaultProps = {
    items: [],
};

export default Campaigns;
