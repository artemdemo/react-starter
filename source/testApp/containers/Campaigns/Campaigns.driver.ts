import { RenderResults } from '../../../__tests__/testRender';
import { testIdExists } from '../../../__tests__/utils';
import { TCampaign } from './Campaigns';

type CampaignOptions = {
  name?: string;
  picture?: string;
  description?: string;
};

export const createCampaignMock = (
  options: CampaignOptions = {}
): TCampaign => {
  const {
    name = 'test campaign',
    picture = 'http://via.placeholder.com/150x150',
    description = 'Est amet voluptate',
  } = options;
  return {
    name,
    picture,
    description,
  };
};

export class CampaignsDriver {
  constructor(public readonly component: RenderResults) {}

  campaignsListExist(): boolean {
    return testIdExists(this.component, 'campaigns-list');
  }

  getCampaignsAmount(): number {
    const { getAllByTestId } = this.component;
    const campaignsEl = getAllByTestId('single-campaign');
    return campaignsEl.length;
  }
}
