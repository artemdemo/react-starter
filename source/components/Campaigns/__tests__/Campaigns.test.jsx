import React from 'react';
import renderer from 'react-test-renderer';
import Campaigns from '../Campaigns';

describe('<Campaigns>', () => {
    it('simple render, without data', () => {
        const tree = renderer.create(
            <Campaigns />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('with data', () => {
        const tree = renderer.create(
            <Campaigns
                items={[
                    {
                        name: 'First campaign',
                        picture: 'http://example.com/img-1.png',
                        description: 'Some description #1',
                    },
                    {
                        name: 'Second campaign',
                        picture: 'http://example.com/img-2.png',
                        description: 'Some description #2',
                    },
                ]}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
