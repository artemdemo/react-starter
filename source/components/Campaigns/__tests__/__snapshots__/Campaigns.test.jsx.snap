// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Campaigns> simple render, without data 1`] = `
<div
  className="row"
/>
`;

exports[`<Campaigns> with data 1`] = `
<div
  className="row"
>
  <div
    className="col-6 col-md-4"
  >
    <h3>
      First campaign
    </h3>
    <img
      alt="First campaign"
      src="http://example.com/img-1.png"
    />
    <p>
      Some description #1
    </p>
  </div>
  <div
    className="col-6 col-md-4"
  >
    <h3>
      Second campaign
    </h3>
    <img
      alt="Second campaign"
      src="http://example.com/img-2.png"
    />
    <p>
      Some description #2
    </p>
  </div>
</div>
`;
